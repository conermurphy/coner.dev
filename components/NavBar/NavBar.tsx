import { useRouter } from 'next/router';
import React from 'react';
import NoScrollLink from '../NoScrollLink/NoScrollLink';

interface IProps {
  isMobile?: boolean;
}

interface CustomLinkProps {
  href: string;
  value: string;
  activeLink: boolean;
}

function CustomLink({ href, value, activeLink }: CustomLinkProps): JSX.Element {
  return (
    <li className="block w-max">
      <span className="font-semibold opacity-75 hover:opacity-100 transition-all ease-in-out duration-150">
        <NoScrollLink href={href}>{value}</NoScrollLink>
      </span>
      <div className={activeLink ? 'border-b-2 border-accent' : ''} />
    </li>
  );
}

export default function NavBar({ isMobile = false }: IProps): JSX.Element {
  const { asPath } = useRouter();

  return (
    <nav aria-label="primary">
      <ul
        className={`flex gap-x-8 flex-wrap gap-y-4 ${
          isMobile ? 'flex-col' : 'flex-row'
        }`}
      >
        <CustomLink href="/" value="Home" activeLink={asPath === '/'} key="/" />
        <CustomLink
          href="/blog"
          value="Blog"
          activeLink={asPath.includes('blog')}
          key="/blog"
        />
        <CustomLink
          href="/newsletter"
          value="Newsletter"
          activeLink={asPath.includes('newsletter')}
          key="/newsletter"
        />
        <CustomLink
          href="/contact"
          value="Contact"
          activeLink={asPath.includes('contact')}
          key="/contact"
        />
      </ul>
    </nav>
  );
}
