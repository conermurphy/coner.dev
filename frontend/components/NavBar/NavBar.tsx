'use client';

import { usePathname } from 'next/navigation';
import React from 'react';
import Link from 'next/link';

interface IProps {
  isMobile?: boolean;
}

interface CustomLinkProps {
  href: string;
  value: string;
  activeLink: boolean;
}

function CustomLink({ href, value, activeLink }: CustomLinkProps): JSX.Element {
  return (
    <li className="block w-max">
      <span className="font-heading text-lg font-extrabold opacity-75 hover:opacity-100 transition-all ease-in-out duration-150">
        <Link href={href}>{value}</Link>
      </span>
      <div className={activeLink ? 'border-b-4 border-brand' : ''} />
    </li>
  );
}

export default function NavBar({ isMobile = false }: IProps): JSX.Element {
  const pathname = usePathname();

  return (
    <nav aria-label="primary">
      <ul
        className={`flex gap-x-8 flex-wrap gap-y-4 justify-center ${
          isMobile ? 'flex-col' : 'flex-row'
        }`}
      >
        <CustomLink
          href="/"
          value="Home"
          activeLink={pathname === '/'}
          key="/"
        />
        <CustomLink
          href="/blog"
          value="Blog"
          activeLink={pathname?.includes('blog') || false}
          key="/blog"
        />
        <CustomLink
          href="/newsletter"
          value="Newsletter"
          activeLink={pathname?.includes('newsletter') || false}
          key="/newsletter"
        />
        <CustomLink
          href="/technical-writing"
          value="Technical Writing"
          activeLink={pathname?.includes('technical-writing') || false}
          key="/technical-writing"
        />
        <CustomLink
          href="/links"
          value="Links"
          activeLink={pathname?.includes('links') || false}
          key="/links"
        />
        <CustomLink
          href="/contact"
          value="Contact"
          activeLink={pathname?.includes('contact') || false}
          key="/contact"
        />
      </ul>
    </nav>
  );
}
