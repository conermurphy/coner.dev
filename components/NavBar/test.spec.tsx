import { render, screen } from '@testing-library/react';
import React from 'react';
import NavBar from './NavBar';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '/',
    };
  },
}));

describe('NavBar', () => {
  it('3 links should be rendered', () => {
    render(<NavBar />);

    const links = screen.queryAllByRole('link').length;
    const homeLink = screen.queryByText(/home/i);
    const blogLink = screen.queryByText(/blog/i);
    const newsletterLink = screen.queryByText(/newsletter/i);
    const contactLink = screen.queryByText(/contact/i);

    expect(links).toEqual(4);
    expect(homeLink).toBeVisible();
    expect(blogLink).toBeVisible();
    expect(newsletterLink).toBeVisible();
    expect(contactLink).toBeVisible();
  });
});
