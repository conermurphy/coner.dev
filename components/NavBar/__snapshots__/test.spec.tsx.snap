// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Matches the snapshot 1`] = `
<div>
  <nav
    class="flex flex-row gap-8"
  >
    <p
      class="font-semibold"
      href="/"
    >
      Home
    </p>
    <p
      class="font-semibold"
      href="/blog"
    >
      Blog
    </p>
    <p
      class="font-semibold"
      href="/contact"
    >
      Contact
    </p>
  </nav>
</div>
`;
